packages/main/src/CalendarLegend.ts

```diff
@@ -0,0 +1,203 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import slot from "@ui5/webcomponents-base/dist/decorators/slot.js";
+import event from "@ui5/webcomponents-base/dist/decorators/event.js";
+import ItemNavigation from "@ui5/webcomponents-base/dist/delegate/ItemNavigation.js";
+import NavigationMode from "@ui5/webcomponents-base/dist/types/NavigationMode.js";
+import {
+	isDown,
+	isUp,
+} from "@ui5/webcomponents-base/dist/Keys.js";
+import ItemNavigationBehavior from "@ui5/webcomponents-base/dist/types/ItemNavigationBehavior.js";
+import CalendarLegendItemType from "./types/CalendarLegendItemType.js";
+import CalendarLegendTemplate from "./generated/templates/CalendarLegendTemplate.lit.js";
+import CalendarLegendItem from "./CalendarLegendItem.js";
+
+// Styles
+import CalendarLegendCss from "./generated/themes/CalendarLegend.css.js";
+
+type CalendarLegendItemSelectionChangeEventDetail = {
+	item: CalendarLegendItem;
+}
+
+/**
+ * @class CalendarLegend
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * The <code>ui5-calendar-legend</code> component is designed for use within the <code>ui5-calendar</code> to display a legend.
+ * Each <code>ui5-calendar-legend-item</code> represents a unique date type, specifying its visual style
+ * and a corresponding textual label.
+ *
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import "@ui5/webcomponents/dist/CalendarLegend.js";</code>
+ *
+ * @slot items - Defines the items of the component.
+ *
+ * @constructor
+ * @extends UI5Element
+ * @public
+ * @since 1.22.0
+ */
+@customElement({
+	tag: "ui5-calendar-legend",
+	renderer: litRender,
+	styles: CalendarLegendCss,
+	template: CalendarLegendTemplate,
+	dependencies: [CalendarLegendItem],
+})
+@event("_calendar-legend-selection-change", {
```

Comment:
Events with details object should use generic in order to provide type for detail object

`@event<CalendarLegendItemSelectionChangeEventDetail>("_calendar-legend-selection-change"....